Doc/library/pathlib.rst

```diff
@@ -1633,29 +1633,9 @@ Copying, renaming and deleting
       Added return value, return the new :class:`!Path` instance.
 
 
-.. method:: Path.unlink(missing_ok=False)
-
-   Remove this file or symbolic link.  If the path points to a directory,
-   use :func:`Path.rmdir` instead.
-
-   If *missing_ok* is false (the default), :exc:`FileNotFoundError` is
-   raised if the path does not exist.
-
-   If *missing_ok* is true, :exc:`FileNotFoundError` exceptions will be
-   ignored (same behavior as the POSIX ``rm -f`` command).
-
-   .. versionchanged:: 3.8
-      The *missing_ok* parameter was added.
-
-
-.. method:: Path.rmdir()
-
-   Remove this directory.  The directory must be empty.
-
-
-.. method:: Path.rmtree(ignore_errors=False, on_error=None)
+.. method:: Path.delete(ignore_errors=False, on_error=None)
```

Comment:
At risk of bikeshedding, perhaps ``remove``? 'Delete' doesn't seem to be a frequently-used term for OS-level operations -- see PowerShell's [`Remove-Item`](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/remove-item?view=powershell-7.4), or [``rm(tree)``](https://man7.org/linux/man-pages/man1/rm.1.html) in Unix. (DOS has ``del``, but PowerShell chose to use 'remove' over 'delete').

We have [`os.remove`](https://docs.python.org/dev/library/os.html#os.remove), so on a synonyms argument you can go both ways -- good in that this is a superset and does what I expect, or bad in that it doesn't exactly mirror ``os.remove``.

If this is part of a grand plan that I am missing though, do ignore me entirely.

A

Reply:
It's a good suggestion, I'll give it some thought.

My initial reaction: I reckon "delete" is more common in GUI file browsers, e.g. my Linux box offers a "Delete" option in the context menu, and macOS offers "Delete Immediately..." for items in the Bin. And I think "delete" is a tiny bit more specific, e.g. consider the difference between *removing* furniture from a room and *deleting* furniture from a room. I suspect that OSs use "remove" because of hard links: `rm some/file` may not delete the file data is multiply hardlinked (also the reason for the `unlink()` name). Given pathlib's beginner-friendly nature it feels OK to me to gloss over the distinction. At least, that's how I feel right now :)

Reply:
I think I'm -1 on `remove()`.

It's true that `pathlib.Path` and `os` have methods and functions that share names, and that in some cases the pathlib methods can do more than just wrap the os functions (`mkdir()` being the most prominent example, with its *parents* and *exist_ok* arguments).

But there's no precedent for diverging as much as `os.remove()` and `pathlib.Path.remove()` would here, and diverging without requiring the user to pass additional arguments. The potential for confusion would be too great!

I do appreciate you sharing your suggestion all the same.

Reply:
FWIW, I've also considered `rm()`. This doesn't suffer from the `remove()` problem, but it's rather cryptic for non-users of unix shells (does it summon the Royal Marines?) and in other users it invites comparisons to the `rm` command that I'd rather avoid, as pathlib isn't in the business of precisely matching shell APIs.

Reply:
> does it summon the Royal Marines?

No task is too hard for the Special Boat Service! Though this would be a fun easter egg, notwithstanding the thoughts and feelings of the First Sea Lord...

> I think I'm -1 on `remove()`.

The argument you lay out is compelling, and especially given ``Path.delete``'s intended use as a one-stop-shop I think makes sense. I remember being very surprised by ``os.unlink`` when I first came across it, and ``delete`` does have the benefit of simplicity.

Thank you for entertaining my notions!